javascript_nodejs/81.facebook-events/bot.js

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityTypes, MessageFactory } = require('botbuilder');
+const { DialogSet, ChoicePrompt } = require('botbuilder-dialogs');
+
+const DIALOG_STATE_PROPERTY = 'dialogState';
+const USER_PROFILE_PROPERTY = 'user';
+const SAMPLE_PROMPT = 'sample_prompt';
+
+/**
+ * A simple bot that responds handles and shows Facebook events.
+ */
+class FacebookEventsBot {
+    /**
+     * 
+     * @param {ConversationState} conversationState A ConversationState object used to store the dialog state.
+     * @param {UserState} userState A UserState object used to store values specific to the user.
+     */
+    constructor (conversationState, userState) {
+
+        this.conversationState = conversationState;
+        this.userState = userState;
+
+        this.dialogState = this.conversationState.createProperty(DIALOG_STATE_PROPERTY);
+
+        this.userProfile = this.userState.createProperty(USER_PROFILE_PROPERTY);
+
+        this.dialogs = new DialogSet(this.dialogState);
+        this.dialogs.add(new ChoicePrompt(SAMPLE_PROMPT))
+    }
+
+    /**
+     * Every conversation turn for our FacebookEventsBot will call this method.
+     * There are no dialogs used, since it's "single turn" processing, meaning a single request and
+     * response, with no stateful conversation.
```

Comment:
This comment doesn't seem right, as I see conversationState, userState, and dialogs all manipulated in the constructor.